importers/wecdis_importer.py

```diff
@@ -0,0 +1,344 @@
+from datetime import datetime
+from enum import Enum
+
+from pepys_import.core.formats import unit_registry
+from pepys_import.core.formats.location import Location
+from pepys_import.core.validators import constants
+from pepys_import.file.highlighter.support.combine import combine_tokens
+from pepys_import.file.importer import Importer
+from pepys_import.utils.unit_utils import convert_absolute_angle, convert_speed
+
+
+class MsgType(str, Enum):
+    """The supported WECDIS message types"""
+
+    TIME = "DZA"
+    PLATFORM = "VNM"
+    CONTACT = "CONTACT"
+    POSITION = "CPOS"
+    TMA = "TMA"
+
+
+class WecdisImporter(Importer):
+    def __init__(self):
+        super().__init__(
+            name="WECDIS File Format Importer",
+            validation_level=constants.BASIC_LEVEL,
+            short_name="WECDIS Importer",
+            datafile_type="WECDIS",
+        )
+
+        self.platform_name = None
+        self.timestamp = None
+
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".LOG" or suffix.upper() == ".TXT"
+
+    def can_load_this_filename(self, filename):
+        return True
+
+    def can_load_this_header(self, header):
+        return "$POSL" in header
+
+    def can_load_this_file(self, file_contents):
+        # Need to differentiate from general NMEA - so check charts/version available
```

Comment:
@mew-nsc - is it worth limiting how many rows of data we merge before we check for chart contents.  I'm pretty sure analysts get WECDIS logs that are 100Mb long.  I guess we can expect the charts to be declared in the first 100 rows.  I can verify when I see the collated samples, next time I'm in.

Reply:
Good point Ian. I've made that change so that I don't get caught out while testing. Would be good to know whether VER/Chart tags are always present in WECDIS files, and whether they always appear early on.